src/seedu/addressbook/data/person/Email.java

```diff
@@ -3,31 +3,27 @@
 import seedu.addressbook.data.exception.IllegalValueException;
 
 /**
- * Represents a Person's email in the address book.
- * Guarantees: immutable; is valid as declared in {@link #isValidEmail(String)}
+ * Represents a Person's email in the address book. Guarantees: immutable; is
+ * valid as declared in {@link #isValidEmail(String)}
  */
-public class Email {
+public class Email extends Contact {
 
     public static final String EXAMPLE = "valid@e.mail";
-    public static final String MESSAGE_EMAIL_CONSTRAINTS =
-            "Person emails should be 2 alphanumeric/period strings separated by '@'";
+    public static final String MESSAGE_EMAIL_CONSTRAINTS = "Person emails should be 2 alphanumeric/period strings separated by '@'";
     public static final String EMAIL_VALIDATION_REGEX = "[\\w\\.]+@[\\w\\.]+";
 
-    public final String value;
-    private boolean isPrivate;
-
     /**
      * Validates given email.
      *
-     * @throws IllegalValueException if given email address string is invalid.
+     * @throws IllegalValueException
+     *             if given email address string is invalid.
      */
     public Email(String email, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
-        email = email.trim();
+        super(email.trim(), isPrivate);
+
```

Comment:
think how you can also push the validation logic to the parent class using the super constructor. :-)